ClientApp/src/components/AddListing.js

```diff
@@ -0,0 +1,70 @@
+import React from "react";
+import { useForm } from "react-hook-form";
+import axios from "axios";
+import { Redirect } from 'react-router';
+
+export default function AddListing() {
+    const { register, handleSubmit, formState } = useForm({ mode: 'onBlur', defaultValues: { Quantity: 1, Location: "Lithuania" }, shouldUseNativeValidation: true });
+    const { isSubmitting, isSubmitted } = formState;
+
+    const onSubmit = data => {
+        var imageAttached = false;
+        const formData = new FormData();
+        if (data.ImagePath.length > 0) {
+            formData.append('FormFile', data.ImagePath[0]);
+            formData.append('FileName', data.ImagePath[0].name);
+            data.ImagePath = 'images/' + data.ImagePath[0].name;
+            imageAttached = true;
+        }
+        else {
+            data.ImagePath = 'images/default.png';
+        }
+        fetch('listing', {
+            mode: 'cors',
+            method: 'POST',
+            headers: { 'Content-type': 'application/json' },
+            body: JSON.stringify(data),
+        }).then(function () {
+            if (imageAttached) {
+                const res = axios.post('image', formData);
+                console.log(res);
+            }
+        }).then(function () {
+        });
```

Comment:
This empty 'then' could be emitted.
Also, (only a suggestion) you could look into async/await.
So for this case it could look something like this:
const result = await fetch(...);
if (result.ok) { axios.post(...) }